crowbar_framework/app/models/ipmi_service.rb

```diff
@@ -71,6 +71,16 @@ def transition(inst, name, state)
       bmc_role = node.admin? ? "bmc-nat-router" : "bmc-nat-client"
       result = add_role_to_instance_and_node("ipmi", inst, name, db, role, bmc_role)
 
+      @logger.debug("IPMI transition: leaving from discovered state for #{name}")
+      a = [200, { name: name }] if result
+      a = [400, "Failed to add role to node"] unless result
+      return a
+    end
+
+    # node got allocated, ensure that node can use BMC network by allocating an
+    # IP / enabling the interface
+    if state == "hardware-installing"
```

Comment:
Don't you need to add "hardware-installing"  to the "transition_list" in the proposal to make the is work?